connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -111,12 +115,29 @@ protected HttpEndpointBuilderFactory.HttpEndpointBuilder setUpBOPEndpoint() {
         }
     }
 
-    private HttpEndpointBuilderFactory.HttpEndpointBuilder setupRecipientResolverEndpoint() {
+    private HttpEndpointBuilder setupRecipientResolverEndpoint() {
         final String fullURL = emailConnectorConfig.getRecipientsResolverServiceURL() + "/internal/recipients-resolver";
+
         if (fullURL.startsWith("https")) {
-            return https(fullURL.replace("https://", ""))
-                    .sslContextParameters(getSslContextParameters())
-                    .x509HostnameVerifier(NoopHostnameVerifier.INSTANCE);
+            HttpEndpointBuilder endpointBuilder = https(fullURL.replace("https://", ""));
+            if (emailConnectorConfig.getRecipientsResolverTrustStorePath().isPresent() && emailConnectorConfig.getRecipientsResolverTrustStorePassword().isPresent() && emailConnectorConfig.getRecipientsResolverTrustStoreType().isPresent()) {
```

Comment:
What about a keystore without password ? is password field is always mandatory, even it it's empty ?
I ask because on Kafka client setup, we must not add the keystore password configuration key if its empty.